In [3]:
import pandas as pd

In [6]:
df = pd.read_csv('/content/hasil_saya (1).csv')

In [ ]:
df

Unnamed: 0                         userName  \
0             0                      Gadri Bandu   
1             1         070_Putri Azzahra Nurdin   
2             2              Novrealita Setiyana   
3             3                       auah gelap   
4             4                      Adi Nugroho   
..          ...                              ...   
995         995                       TEGUH TEAM   
996         996                         Zen Flow   
997         997                     Edward Gosal   
998         998                        Zaaa Qirr   
999         999  Andik Naufal Gaming from Nabila   

                                               content  score  
0    Saya kasih bintang tiga dulu soalnga gemenya b...      3  
1    Bagus bgtt. Dari segi story, grafik sampe puzz...      5  
2    Sejauh ini aku main aku suka banget sama model...      5  
3    Semuanya sudah sempurna tapi Sangat disayangka...      3  
4    Grafik bagus,story bagus,terutama archon quest...      5  
..                                                 ...    ...  
995  Masih bingung sih dengan cara game ini ngerend...      5  
996  game nya seru dari segi story. saran buat Hoyo...      5  
997  Game ini sangat bagus dengan map yg sangat lua...      5  
998  Sangat seru dan asik untuk dimainkan grafik hd...      5  
999  Memuat game terlalu lama . Mau masuk game aja ...      2  

[1000 rows x 4 columns]

In [7]:
from sklearn.feature_extraction.text import CountVectorizer, TfidfTransformer, TfidfVectorizer

In [8]:
text = df['content']

In [9]:
text

0      Saya kasih bintang tiga dulu soalnga gemenya b...
1      Bagus bgtt. Dari segi story, grafik sampe puzz...
2      Sejauh ini aku main aku suka banget sama model...
3      Semuanya sudah sempurna tapi Sangat disayangka...
4      Grafik bagus,story bagus,terutama archon quest...
                             ...                        
995    Masih bingung sih dengan cara game ini ngerend...
996    game nya seru dari segi story. saran buat Hoyo...
997    Game ini sangat bagus dengan map yg sangat lua...
998    Sangat seru dan asik untuk dimainkan grafik hd...
999    Memuat game terlalu lama . Mau masuk game aja ...
Name: content, Length: 1000, dtype: object

In [10]:
tfidf = TfidfVectorizer()
tfidf_matrix = tfidf.fit_transform(text)
print(tfidf_matrix)

  (0, 3251)	0.1055299758616986
  (0, 2602)	0.14500522566335713
  (0, 538)	0.18749785600261215
  (0, 340)	0.22117246882497477
  (0, 953)	0.05838791659996207
  (0, 1680)	0.06532245345690199
  (0, 2368)	0.12440423861415989
  (0, 1785)	0.135243996694897
  (0, 3502)	0.12713065282236005
  (0, 1862)	0.09833987824244249
  (0, 1341)	0.1718400527197851
  (0, 639)	0.12951658229423796
  (0, 3145)	0.12312688906052045
  (0, 3440)	0.1718400527197851
  (0, 2802)	0.15059373755015762
  (0, 1915)	0.3998523073106945
  (0, 370)	0.19992615365534724
  (0, 443)	0.22117246882497477
  (0, 870)	0.058161701894011326
  (0, 2147)	0.16625154083298463
  (0, 428)	0.08481675158152252
  (0, 2560)	0.17867983848571975
  (0, 3355)	0.20874417117223965
  (0, 4099)	0.1756559033334547
  (0, 2933)	0.05691042801487717
  :	:
  (997, 1364)	0.06318278374092187
  (997, 377)	0.14013407953934043
  (998, 329)	0.472267532819263
  (998, 1558)	0.472267532819263
  (998, 1058)	0.472267532819263
  (998, 3674)	0.2908434965194849
  (998, 3488)

# Interpretasi
Format "(i, j) x.y" digunakan untuk menunjukkan skor TF-IDF suatu istilah dalam suatu dokumen. Mari kita jelaskan interpretasi yang lebih terperinci:

Contoh "(0, 3251) 0.1055299758616986": (pada baris 1)

"(0, 3251)" mengindikasikan bahwa ini adalah skor TF-IDF untuk istilah yang terdapat di dokumen 0 dan memiliki indeks 3251 dalam representasi fitur.
"0.1055299758616986" adalah nilai skor TF-IDF untuk istilah tersebut dalam dokumen 0. Semakin tinggi nilai ini, semakin penting atau khas istilah tersebut dalam dokumen tersebut.
Dengan kata lain, output ini memberikan informasi tentang seberapa penting atau berbedanya suatu istilah dalam setiap dokumen. Semakin tinggi skor TF-IDF suatu istilah dalam suatu dokumen, semakin khas atau penting istilah tersebut dalam dokumen tersebut. Ini membantu dalam proses pengambilan informasi, mengidentifikasi istilah-istilah yang memiliki dampak signifikan dalam setiap dokumen.







In [16]:
nama_fitur=tfidf.get_feature_names_out()
nama_fitur_baru=pd.DataFrame(nama_fitur)
nama_fitur_baru

0
0              00
1             000
2           000an
3        08102020
4     0sebelumnya
...           ...
4348        yutub
4349        zaman
4350        zhong
4351      zhongli
4352         zonk

[4353 rows x 1 columns]

# Interpretasi
data pada tabel di atas adalah fitur-fitur/kata-kata yang terdapat dalam dokumen tersebut. Saya tampilkan dengan tabel agar bisa dilihat semua.


In [14]:
pd.DataFrame(tfidf_matrix.toarray(), columns=tfidf.get_feature_names_out(), index=text)

00  000  000an  08102020  \
content                                                                         
Saya kasih bintang tiga dulu soalnga gemenya ba...  0.0  0.0    0.0       0.0   
Bagus bgtt. Dari segi story, grafik sampe puzzl...  0.0  0.0    0.0       0.0   
Sejauh ini aku main aku suka banget sama model ...  0.0  0.0    0.0       0.0   
Semuanya sudah sempurna tapi Sangat disayangkan...  0.0  0.0    0.0       0.0   
Grafik bagus,story bagus,terutama archon quest,...  0.0  0.0    0.0       0.0   
...                                                 ...  ...    ...       ...   
Masih bingung sih dengan cara game ini ngerende...  0.0  0.0    0.0       0.0   
game nya seru dari segi story. saran buat Hoyov...  0.0  0.0    0.0       0.0   
Game ini sangat bagus dengan map yg sangat luas...  0.0  0.0    0.0       0.0   
Sangat seru dan asik untuk dimainkan grafik hd ...  0.0  0.0    0.0       0.0   
Memuat game terlalu lama . Mau masuk game aja l...  0.0  0.0    0.0       0.0   

                                                    0sebelumnya   10  100  \
content                                                                     
Saya kasih bintang tiga dulu soalnga gemenya ba...          0.0  0.0  0.0   
Bagus bgtt. Dari segi story, grafik sampe puzzl...          0.0  0.0  0.0   
Sejauh ini aku main aku suka banget sama model ...          0.0  0.0  0.0   
Semuanya sudah sempurna tapi Sangat disayangkan...          0.0  0.0  0.0   
Grafik bagus,story bagus,terutama archon quest,...          0.0  0.0  0.0   
...                                                         ...  ...  ...   
Masih bingung sih dengan cara game ini ngerende...          0.0  0.0  0.0   
game nya seru dari segi story. saran buat Hoyov...          0.0  0.0  0.0   
Game ini sangat bagus dengan map yg sangat luas...          0.0  0.0  0.0   
Sangat seru dan asik untuk dimainkan grafik hd ...          0.0  0.0  0.0   
Memuat game terlalu lama . Mau masuk game aja l...          0.0  0.0  0.0   

                                                    1000  100ms  107  ...  \
content                                                               ...   
Saya kasih bintang tiga dulu soalnga gemenya ba...   0.0    0.0  0.0  ...   
Bagus bgtt. Dari segi story, grafik sampe puzzl...   0.0    0.0  0.0  ...   
Sejauh ini aku main aku suka banget sama model ...   0.0    0.0  0.0  ...   
Semuanya sudah sempurna tapi Sangat disayangkan...   0.0    0.0  0.0  ...   
Grafik bagus,story bagus,terutama archon quest,...   0.0    0.0  0.0  ...   
...                                                  ...    ...  ...  ...   
Masih bingung sih dengan cara game ini ngerende...   0.0    0.0  0.0  ...   
game nya seru dari segi story. saran buat Hoyov...   0.0    0.0  0.0  ...   
Game ini sangat bagus dengan map yg sangat luas...   0.0    0.0  0.0  ...   
Sangat seru dan asik untuk dimainkan grafik hd ...   0.0    0.0  0.0  ...   
Memuat game terlalu lama . Mau masuk game aja l...   0.0    0.0  0.0  ...   

                                                    yoimiya  you  youtube  \
content                                                                     
Saya kasih bintang tiga dulu soalnga gemenya ba...      0.0  0.0      0.0   
Bagus bgtt. Dari segi story, grafik sampe puzzl...      0.0  0.0      0.0   
Sejauh ini aku main aku suka banget sama model ...      0.0  0.0      0.0   
Semuanya sudah sempurna tapi Sangat disayangkan...      0.0  0.0      0.0   
Grafik bagus,story bagus,terutama archon quest,...      0.0  0.0      0.0   
...                                                     ...  ...      ...   
Masih bingung sih dengan cara game ini ngerende...      0.0  0.0      0.0   
game nya seru dari segi story. saran buat Hoyov...      0.0  0.0      0.0   
Game ini sangat bagus dengan map yg sangat luas...      0.0  0.0      0.0   
Sangat seru dan asik untuk dimainkan grafik hd ...      0.0  0.0      0.0   
Memuat game terlalu lama . Mau masuk game aja l...      0.0  0.0      0

# Interpretasi
Angka-angka tiap kolom yang kebanyakan bernilai 0 adalah akibat tidak adanya representasi kata tersebut pada kalimat yang ditampilkan. Kolom yang dihasilkan juga terlalu banyak karena pada kasus ini data belum dibersihkan sehingga banyak noise seperti '00','000', '0sebelumnya', dll.
Seharusnya jika kita dapat melihat semua fitur/kolom yang ada maka kata-kata yang muncul pada kalimat pertama akan memiliki nilai pada kolom kata tersebut, nilainya didasarkan dengan rumus TF-IDF yaitu:
**TF-IDF(t, d) = TF(t, d) * IDF(t)**,
dengan

**TF(t, d) = (jumlah kemunculan kata t dalam dokumen d) / (jumlah kata total dalam dokumen d)**,
dan
**IDF(t) = log(N / (jumlah dokumen yang mengandung kata t))**